buildlib/azure-pipelines-pr.yml

```diff
@@ -47,6 +47,26 @@ stages:
                  echo "##vso[task.complete result=Failed;]"
               fi
             condition: eq(variables['Build.Reason'], 'PullRequest')
+  - stage: Format
+    jobs:
+      # Check that the code is formatted according to the code style guidelines
+      - job: format
+        displayName: format code
+        container: fedora
+        steps:
+          - checkout: self
+            clean: true
+
+          - bash: |
+              git-format-diff ${SYSTEM_PULLREQUEST_TARGETBRANCH}
```

Comment:
what if AZP added merge commit?
need to use `"remotes/origin/$(System.PullRequest.TargetBranch)..$(Build.SourceVersion)"`

Reply:
Replaced `TARGETBRANCH` with `remotes/origin/$(SYSTEM_PULLREQUEST_TARGETBRANCH)` - I think it's supposed to be good enough?

Reply:
the current head in git can also contain merge commit, which we dont want to check

Reply:
Then maybe it would be better to skip the format check at all in such case (if there's a merge commit - it means that format check has already been done).

Reply:
azp always adds merge commit with master since it's not fast-forward-only project
you can add `git log --oneline -10` to the AZP script to see it

Reply:
Understood. But do we need to check its format? Wouldn't that be a double check, after the PR commit was already checked?